static/js/publisher-pages/pages/Builds/RepoConnected.tsx

```diff
@@ -0,0 +1,314 @@
+import { useRecoilValue, useSetRecoilState } from "recoil";
+import { useParams, Link } from "react-router-dom";
+import { formatDistanceToNow, formatDuration } from "date-fns";
+import {
+  Strip,
+  Button,
+  Modal,
+  Row,
+  Col,
+  MainTable,
+} from "@canonical/react-components";
+
+import { buildRepoConnectedState, githubDataState } from "../../state/atoms";
+
+function RepoConnected({
+  autoTriggerBuild,
+  setAutoTriggerBuild,
+}: {
+  autoTriggerBuild: boolean;
+  setAutoTriggerBuild: Function;
+}): JSX.Element {
+  const { snapId } = useParams();
+  const setRepoConnected = useSetRecoilState(buildRepoConnectedState);
+  const githubData = useRecoilValue(githubDataState);
+  const [disconnecting, setDisconnecting] = useState<boolean>(false);
+  const [disconnectModalOpen, setDisconnectModalOpen] =
+    useState<boolean>(false);
+  const [triggeringBuild, setTriggeringBuild] = useState<boolean>(false);
+  const { isLoading, isFetched, data, refetch } = useQuery({
+    queryKey: ["repo"],
+    queryFn: async () => {
+      const response = await fetch(`/api/${snapId}/builds`);
+      const data = await response.json();
+      return data.data;
+    },
+    onSettled: () => {
+      setTriggeringBuild(false);
+    },
+  });
+
+  const formatDurationString = (duration?: string): string => {
+    if (!duration) {
+      return "-";
+    }
+
+    const durationParts = duration.split(":");
+
+    return formatDuration({
+      hours: parseInt(durationParts[0]),
+      minutes: parseInt(durationParts[1]),
+      seconds: Math.floor(parseInt(durationParts[2])),
+    });
+  };
+
+  const formatStatus = (status: string): JSX.Element => {
+    switch (status) {
+      case "never_built":
+        return <>Never built</>;
+      case "building_soon":
+        return <>Building soon</>;
+      case "wont_release":
+        return <>Won't release</>;
+      case "released":
+        return <>Released</>;
+      case "release_failed":
+        return <>Release failed</>;
+      case "releasing_soon":
+        return <>Releasing soon</>;
+      case "in_progress":
+        return (
+          <>
+            <i className="p-icon--spinner u-animation--spin" />
+            In progress
+          </>
+        );
+      case "failed_to_build":
+        return <>Failed to build</>;
+      case "cancelled":
+        return <>Cancelled</>;
+      case "unknown":
+        return <>Unknown</>;
+      case "ERROR":
+        return <>Error</>;
+      case "SUCCESS":
+        return <>Success</>;
+      case "IDLE":
+        return <>Idle</>;
+      default:
+        return <>{status}</>;
+    }
+  };
+
+  async function trigger() {
+    const response = await fetch(`/api/${snapId}/builds/trigger-build`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-CSRFToken": window.CSRF_TOKEN,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error("Unable to trigger new build");
```

Comment:
How do we handle the thrown errors? Do we have an ErrorBoundary wrapped the components?

Reply:
We don't have a standard way of handling errors across our projects. This is something we should discuss further as a team I think.